src/openms/source/CHEMISTRY/ProForma.cpp

```diff
@@ -197,10 +247,10 @@ namespace OpenMS
       sequence_ += AASequence::fromString(range); // Append the range to the sequence
 
       // Apply the modification to the entire range
-      ModificationAttributes attributes;
-      attributes.modification_name = modification;
-      attributes.range = {range_seq_start, range_seq_start + range.size() - 1};  // Store the range as a pair (start, end)
-      OPENMS_LOG_DEBUG << "Range of modification: " << attributes.range.first << " - " << attributes.range.second << std::endl;
+      ResidueModification attributes;
+      attributes.setName(modification);
+      attributes.setPositionRange(range_seq_start, range_seq_start + range.size() - 1);  // Store the range as a pair (start, end)
```

Comment:
From a OOP point of view this doesn’t look like the range should be part of ResidueModification.